drivers/sensor/max30208/max30208.c

```diff
@@ -0,0 +1,329 @@
+/*
+ * Copyright (c) 2021, Silvano Cortesi
+ * Copyright (c) 2021, ETH Zürich
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT maxim_max30208
+
+#include <sys/byteorder.h>
+#include <logging/log.h>
+
+#include "max30208.h"
+
+LOG_MODULE_REGISTER(MAX30208, CONFIG_SENSOR_LOG_LEVEL);
+
+static int max30208_start_measurement(const struct device *dev)
+{
+	const struct max30208_config *config = dev->config;
+
+	/* Trigger one measurement */
+	int ret = i2c_reg_write_byte_dt(&config->bus, MAX30208_REG_TEMP_SETUP,
+					MAX30208_CONVERT_T_MASK);
+
+	if (ret < 0) {
+		return ret;
+	}
+
+	return 0;
+}
+
+static int max30208_channel_get(const struct device *dev, enum sensor_channel chan,
+				struct sensor_value *val)
+{
+	struct max30208_data *data = dev->data;
+	uint8_t local_buffer[MAX30208_BYTES_PER_VAL] = { 0 };
+
+	if (chan != SENSOR_CHAN_AMBIENT_TEMP) {
+		return -ENOTSUP;
+	}
+
+	/* If new value in ring buffer available, get it */
+	if (ring_buf_size_get(&data->raw_buffer) < MAX30208_BYTES_PER_VAL) {
+		return -ENODATA;
+	}
+
+	int ret = ring_buf_get(&data->raw_buffer, local_buffer, MAX30208_BYTES_PER_VAL);
```

Comment:
this function returns `uint32_t`, so error handling here is incorrect. ref https://docs.zephyrproject.org/latest/reference/data_structures/ring_buffers.html#c.ring_buf_get